gc/base/MemoryPoolAddressOrderedListBase.cpp

```diff
@@ -168,3 +168,34 @@ MM_MemoryPoolAddressOrderedListBase::abandonMemoryInPool(MM_EnvironmentBase* env
 	abandonHeapChunk((MM_HeapLinkedFreeHeader*)address, (uint8_t*)address + size);
 }
 
+#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+uintptr_t
+MM_MemoryPoolAddressOrderedListBase::releaseMemoryPoolFreeEntryMemoryPages(MM_EnvironmentBase* env, MM_HeapLinkedFreeHeader* freeEntry)
+{
+	uintptr_t freedMemory = 0;
+	MM_HeapLinkedFreeHeader* currentFreeEntry = freeEntry;
+	uintptr_t pageSize = env->getExtensions()->heap->getPageSize();
+	while (NULL != currentFreeEntry) {
+		if (currentFreeEntry->getSize() >= ((2 * pageSize) + sizeof(MM_HeapLinkedFreeHeader))) {
+			/* release/decommit memory after Header */
+			uint8_t* addressBase = (uint8_t*)MM_Math::roundToCeiling(pageSize, (uintptr_t)((uint8_t*)currentFreeEntry + sizeof(MM_HeapLinkedFreeHeader)));
+			uintptr_t totalFreePagesCount = (currentFreeEntry->getSize() - (uintptr_t)(addressBase - (uint8_t*)currentFreeEntry)) / pageSize;
+			if (1 < totalFreePagesCount) {
+				uintptr_t commitPagesCount = (0 < _extensions->idleMinimumFree) ? (totalFreePagesCount * _extensions->idleMinimumFree / 100) : 0;
+				uintptr_t decommitPagesCount = totalFreePagesCount - commitPagesCount;
+
+				/* leave commited pages of memory aside header */
+				addressBase += commitPagesCount * pageSize;
+				/* now decommit pages of memory */
+				if (0 < decommitPagesCount) {
+					if (_extensions->heap->decommitMemory(addressBase, decommitPagesCount * pageSize, NULL, currentFreeEntry->afterEnd())) {
+						freedMemory += decommitPagesCount * pageSize;
+					}
+				}
+			}
+		}
+		currentFreeEntry = currentFreeEntry->getNext();
+	}
+	return freedMemory;
+}
+#endif
```

Comment:
To be consistent with the method name, the return variable should be releasedMemory.